src/main/java/core/basesyntax/service/FruitTransactionServiceImpl.java

```diff
@@ -0,0 +1,34 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operation.OperationStrategy;
+import java.util.List;
+
+public class FruitTransactionServiceImpl implements FruitTransactionService {
+    private FruitDao fruitDao;
+    private OperationStrategy operationStrategy;
+
+    public FruitTransactionServiceImpl(FruitDao fruitDao, OperationStrategy strategy) {
+        this.fruitDao = fruitDao;
+        operationStrategy = strategy;
+    }
+
+    @Override
+    public void createNewFruitTransaction(String type, String fruit, int quantity) {
+        FruitTransaction.Operation operation = operationStrategy.get(type).getOperation();
+        FruitTransaction fruitTransaction = new FruitTransaction(operation, fruit, quantity);
+        fruitDao.add(fruitTransaction);
+    }
```

Comment:
this method/class should be about parsing. It shouldn't be concerned with what happens to the transaction that it creates after completion - it's SRP violation, don't do two things at once. Just return it.